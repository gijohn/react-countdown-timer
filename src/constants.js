export const START_COUNTDOWN_TIMER = 'startCountDownTimer';
export const STOP_COUNTDOWN_TIMER = 'stopCountDownTimer';
export const PAUSE_COUNTDOWN_TIMER = 'pauseCountDownTimer';
export const RESUME_COUNTDOWN_TIMER = 'resumtCountDownTimer';
export const LAP_COUNTDOWN_TIMER = 'lapCountDownTimer';
export const MERGE_LAST_LAP = 'mergeLastLap';
export const INITIALIZE_TIMER = 'initializeTimer';

export const STATUS_STOPPED = 'stopped';
export const STATUS_RUNNING = 'started';
export const STATUS_PAUSED = 'paused';

export const SET_TIMER_VALUE = 'setTimerValue';
export const UPDATE_THRESHOLD = 'updateThreshold';