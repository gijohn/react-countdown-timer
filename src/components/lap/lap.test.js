import React from 'react';
import configureMockStore from 'redux-mock-store'
import {STATUS_PAUSED, STATUS_RUNNING, STATUS_STOPPED} from "../../constants";
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Lap from './index';

const mockStore = configureMockStore();

describe('<Lap /> tests', () => {
    it('snapshot test on stopped state', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_STOPPED,
            laps: []
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('snapshot test on stopped state with laps', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_STOPPED,
            laps: [
                {
                    startTime: 4000,
                    endTime: 3000
                },
                {
                    startTime: 3000,
                    endTime: 2000
                }
            ]
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('snapshot test on running state', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_RUNNING,
            laps: []
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('snapshot test on running state with laps', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_RUNNING,
            laps: [
                {
                    startTime: 4000,
                    endTime: 3000
                },
                {
                    startTime: 3000,
                    endTime: 2000
                },
                {
                    startTime: 2000
                }
            ]
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('snapshot test on paused state', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_PAUSED,
            laps: []
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('snapshot test on paused state with laps', () => {
        const store = mockStore({
            threshold: 5000,
            status: STATUS_PAUSED,
            laps: [
                {
                    startTime: 4000,
                    endTime: 3000
                },
                {
                    startTime: 3000,
                    endTime: 2000
                },
                {
                    startTime: 2000
                }
            ]
        });

        const component = renderer.create(
            <Provider store={store}>
                <Lap />
            </Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});