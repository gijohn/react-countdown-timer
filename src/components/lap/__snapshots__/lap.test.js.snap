// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lap /> tests snapshot test on paused state 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`<Lap /> tests snapshot test on paused state with laps 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
      >
        <td>
          3
        </td>
        <td>
          00:00:02.000
        </td>
        <td>
          
        </td>
        <td>
          NaN:NaN:NaN.NaN
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          2
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:02.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          1
        </td>
        <td>
          00:00:04.000
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Lap /> tests snapshot test on running state 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`<Lap /> tests snapshot test on running state with laps 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
      >
        <td>
          3
        </td>
        <td>
          00:00:02.000
        </td>
        <td>
          
        </td>
        <td>
          NaN:NaN:NaN.NaN
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          2
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:02.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          1
        </td>
        <td>
          00:00:04.000
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Lap /> tests snapshot test on stopped state 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`<Lap /> tests snapshot test on stopped state with laps 1`] = `
<div
  className="flex flex-col laps-container"
>
  <div
    className="lap-header"
  >
    Laps
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Lap#
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Elapsed Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
      >
        <td>
          1
        </td>
        <td>
          00:00:04.000
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          2
        </td>
        <td>
          00:00:03.000
        </td>
        <td>
          00:00:02.000
        </td>
        <td>
          00:00:01.000
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
